File: android-app/src/main/java/com/medziku/plantwatering/data/parts/VoltageSensorImpl.java
package com.medziku.plantwatering.data.parts;

import android.os.Parcel;
import android.os.Parcelable;

import com.medziku.plantwatering.station.parts.VoltageSensor;


public class VoltageSensorImpl extends VoltageSensor implements Parcelable {

    public VoltageSensorImpl(String name, float voltage) {
        super(name, voltage);
    }

    private VoltageSensorImpl(Parcel in) {
        super(in.readString(), in.readFloat());
    }

    public static final Creator<VoltageSensorImpl> CREATOR = new Creator<VoltageSensorImpl>() {
        @Override
        public VoltageSensorImpl createFromParcel(Parcel in) {
            return new VoltageSensorImpl(in);
        }

        @Override
        public VoltageSensorImpl[] newArray(int size) {
            return new VoltageSensorImpl[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(getName());
        dest.writeFloat(getVoltage());
    }
}

File: android-app/src/main/kotlin/com/medziku/plantwatering/data/parts/VoltageSensorImpl.kt
package com.medziku.plantwatering.data.parts

import android.os.Parcel
import android.os.Parcelable

import com.medziku.plantwatering.station.parts.VoltageSensor


class VoltageSensorImpl : VoltageSensor, Parcelable {

    constructor(name: String, voltage: Float) : super(name, voltage)

    private constructor(`in`: Parcel) : super(`in`.readString(), `in`.readFloat())

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(name)
        dest.writeFloat(voltage)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<VoltageSensorImpl> = object : Parcelable.Creator<VoltageSensorImpl> {
            override fun createFromParcel(`in`: Parcel): VoltageSensorImpl {
                return VoltageSensorImpl(`in`)
            }

            override fun newArray(size: Int): Array<VoltageSensorImpl?> {
                return arrayOfNulls(size)
            }
        }
    }
}
